File: String/2390.Removing Stars From a String/Solution.java
class Solution {
    public String removeStars(String s) {
        String ans = "";
        int cnt = 0;
        for (int i = s.length() - 1; i > -1; i--) {
            if (s.charAt(i) == '*') {
                cnt++;
            } else {
                if (cnt == 0) {
                    ans = s.charAt(i) + ans;
                } else {
                    cnt--;
                }
            }
        }
        return ans;
    }
}

File: String/2390.Removing Stars From a String/Solution.py
class Solution:
    def removeStars(self, s: str) -> str:
        ans = ""
        cnt = 0
        for i in range(len(s) - 1, -1, -1):
            c = s[i]
            if c == '*':
                cnt += 1
            else:
                if cnt == 0:
                    ans = c + ans
                else:
                    cnt -= 1
        return ans